packages/react-charting/src/components/Legends/Legends.types.ts

```diff
@@ -220,10 +215,28 @@ export interface ILegendsProps {
 
   /**
    * Defines the function that is executed upon hiding of hover card
-   * make sure to send prop  when  the prop  is canSelectMultipleLegends is set to ture
-   * and empty the selecetd state legends
+   * make sure to send prop  when  the prop  is canSelectMultipleLegends is set to true
+   * and empty the selected state legends
    */
   onLegendHoverCardLeave?: VoidFunction;
+
+  /**
+   * Callback issued when the selected option changes.
+   */
+  onChange?: (selectedLegends: string[]) => void;
+
+  /**
+   * Keys (title) that will be initially used to set selected items.
+   * This prop is used for multiSelect scenarios.
+   * In other cases, defaultSelectedLegend should be used.
+   */
+  defaultSelectedLegends?: string[];
+
+  /**
+   * Key that will be initially used to set selected item.
+   * This prop is used for singleSelect scenarios.
+   */
+  defaultSelectedLegend?: string;
```

Comment:
Aren't the `canSelectMultipleLegends` and `defaultSelectedLegends` props sufficient? If only single selection is allowed, we can set the first legend in the array as the `selectedLegend`.
In future, we may use only one state to keep track of selected legends, making this prop redundant.

Reply:
It's another implementation.
I cannot say which one is better.
The 2-properties implementation is borrowed from the Combox interface.
The 1-property implementation is more stable.

@AtishayMsft @krkshitij do you prefer this 1-property props?

Reply:
Lets go with 2 properties - one for single select and 1 for multi select. It would be easier for the consumer to use as multiselection is a relatively less common scenario applicable for line chart only right now.